proto/autoscaling/kubernetes/recommender.proto

```diff
@@ -0,0 +1,72 @@
+syntax = "proto3";
+
+option go_package = "github.com/DataDog/agent-payload/v5/autoscaling/kubernetes";
+
+package datadog.autoscaling.kubernetes;
+
+import "proto/autoscaling/kubernetes/common.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/struct.proto";
+import "proto/deps/github.com/chrusty/protoc-gen-jsonschema/options.proto";
+
+// WorkloadValuesList represents a list of workload recommendation requests
+message WorkloadRecommendationsRequest {
+  option (protoc.gen.jsonschema.message_options).disallow_additional_properties = true;
+
+  repeated WorkloadRecommendationRequest requests  = 1;
+}
+
+// WorkloadRecommendationRequest represents a recommendation request
+message WorkloadRecommendationRequest {
+  option (protoc.gen.jsonschema.message_options).disallow_additional_properties = true;
+  option (protoc.gen.jsonschema.message_options).ignore = true;
+
+  WorkloadTargetRef targetRef = 1 [(protoc.gen.jsonschema.field_options).required = true]; // TargetRef is the target reference of the workload
+
+  WorkloadRecommendationConstraints     constraints = 2 [(protoc.gen.jsonschema.field_options).required = true];
+  WorkloadState                         state       = 3;
+  repeated WorkloadRecommendationTarget targets     = 4;
+  map<string, google.protobuf.Value>    settings    = 5;
+}
+
+// WorkloadRecommendationConstraints represents the constraints for a workload
+message WorkloadRecommendationConstraints {
+  option (protoc.gen.jsonschema.message_options).disallow_additional_properties = true;
+  option (protoc.gen.jsonschema.message_options).ignore = true;
+
+  int32 minReplicas = 1 [(protoc.gen.jsonschema.field_options).required = true]; // MinReplicas is the minimum number of replicas the workload should have
+  int32 maxReplicas = 2 [(protoc.gen.jsonschema.field_options).required = true]; // MaxReplicas is the maximum number of replicas the workload should have
+}
+
+// WorkloadRecommendationTarget represents a target value for a workload
+message WorkloadRecommendationTarget {
+  option (protoc.gen.jsonschema.message_options).disallow_additional_properties = true;
+  option (protoc.gen.jsonschema.message_options).ignore = true;
+
+  string type       = 1 [(protoc.gen.jsonschema.field_options).required = true]; // Type defines the target value type (e.g. CPU, Memory, custom, etc.)
+  int64 targetValue = 2; // TargetValue is the target value of the target
+  int64 lowerBound  = 3; // LowerBound is the lower bound of the target
+  int64 upperBound  = 4; // UpperBound is the upper bound of the target
+}
+
+// WorkloadRecommendationsReply represents a list of recommendations
+message WorkloadRecommendationsReply {
+  option (protoc.gen.jsonschema.message_options).disallow_additional_properties = true;
+
+  Error error = 1; // Error is the error message if new values cannot be calculated
+
+  repeated WorkloadRecommendationReply recommendations = 2; // Recommendations is the list of recommendations
```

Comment:
Maybe worth explaining that the order is the same as the order in requests ? And to avoid issues to we want to have the target in the reply ?